public/js/test/mochitest/browser_dbg-editor-gutter.js

```diff
@@ -0,0 +1,66 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+// Tests the breakpoint gutter and making sure breakpoint icons exist
+// correctly
+
+const TAB_URL = EXAMPLE_URL + "doc-scripts.html";
+
+// Utilities for interacting with the editor
+function clickGutter(dbg, line) {
+  const doc = dbg.win.document;
+  return EventUtils.synthesizeMouseAtCenter(
+    doc.querySelectorAll(".CodeMirror-code .CodeMirror-linenumber")[line - 1],
+    {},
+    dbg.win
+  );
+}
+
+function getLineEl(dbg, line) {
+  const lines = dbg.win.document.querySelectorAll(".CodeMirror-code > div");
+  return lines[line - 1];
+}
+
+function assertEditorBreakpoint(dbg, line, shouldExist) {
+  const exists = !!getLineEl(dbg, line).querySelector(".new-breakpoint");
+  ok(exists === shouldExist,
+     "Breakpoint " + (shouldExist ? "exists" : "does not exist") +
+     " on line " + line);
+}
+
+add_task(function* () {
+  const dbg = yield initDebugger(TAB_URL, "simple1.js");
+  const { selectors: { getBreakpoints, getBreakpoint }, getState } = dbg;
+  const source = findSource(dbg, "simple1.js");
+
+  yield selectSource(dbg, source.url);
+
+  // Make sure that clicking the gutter creates a breakpoint icon.
+  clickGutter(dbg, 4);
+  yield waitForDispatch(dbg, "ADD_BREAKPOINT");
```

Comment:
this is a nit, but would waiting for the "RESUMED" dispatch be similar, but apply for both adding and removing breakpoints.

I bring this up because I like the patten of yielding for async actions. This is a small detail though.